### PR TITLE
Add Makefile command to run checkbashisms on all /bin/sh scripts

### PR DESCRIPTION
**Pull request by gdevenyi**
Based on the shellcheck make target, add a target which checks
for violations of POSIX standards for shell scripts

Signed-off-by: Gabriel A. Devenyi <gdevenyi@gmail.com>

### Motivation and Context

Motivation from #10510 which reports issues with POSIX shells

### Description

Created a modified version of the make shellcheck call to check for bashisms in scripts

### How Has This Been Tested?

Tested locally on Ubuntu 18.04

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [X] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [X] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by gdevenyi** [comment]
This commit will trigger a build failure due to checkbashisms failing.

The current implementation of zfs-functions.sh triggers a bug in checkbashisms which throws an error:
```
error: ./cmd/zed/zed.d/zed-functions.sh: Unterminated quoted string found, EOF reached. Wanted: <">, opened in line 536
```

From my quick check its not a bug.

Full output attached:
[checkbashisms.log](https://github.com/openzfs/zfs/files/4847873/checkbashisms.log)

**Comment by behlendorf** [comment]
I'm a little confused given the work in #10512.   Shouldn't extending the `shellcheck` target to all of the scripts (eventually) allow us to catch the bash'isms and keep new ones out?

**Comment by gdevenyi** [comment]
> I'm a little confused given the work in #10512. Shouldn't extending the shellcheck target to all of the scripts (eventually) allow us to catch the bash'isms and keep new ones out?

I'm not sure whether shellcheck will catch everything, so I intend to keep this tooling around to check the scripts after resolving the shellcheck, they may be complementary.

**Comment by behlendorf** [comment]
It'll be interesting to see what each tool catches.  My understanding is `shellcheck` does a good job of this, but we shall see!  Keeping both targets should be fine as long as we get some utility out of each linter.

File: Makefile.am [hunk 1]
```diff
@@ -146,6 +146,17 @@ shellcheck:
 		echo "skipping shellcheck because shellcheck is not installed"; \
 	fi
 
+PHONY += checkbashisms
+checkbashisms:
+	@if type checkbashisms > /dev/null 2>&1; then \
+		checkbashisms -n -p -x \
+			$$(find ${top_srcdir} -name '.git' -prune -o -name 'build' -prune -o \
+			-type f ! -name 'config*' \
+			-exec bash -c 'awk "NR==1 && /\#\!.*bin\/sh.*/ {print FILENAME;}" "{}"' \;); \
```
Comment (behlendorf) [inline Makefile.am:155]: We're going to want to additionally exclude the following autotools provided scripts which we haven't modified.  These are the only pure scripts in the `config` directory so it's probably cleanest to exclude the whole directory.  

```
config/install-sh
config/depcomp
config/missing
config/config.sub
config/config.guess
config/ltmain.sh
config/compile
```

Comment (gdevenyi) [reply Makefile.am:155]: :+1: I did this in the other PR but I hadn't updated this one yet.

Comment (gdevenyi) [reply Makefile.am:160]: Fixed.

**Pull request merged**